lsp-pyright.el

```diff
@@ -159,35 +183,49 @@ Current LSP WORKSPACE should be passed in."
 (lsp-register-custom-settings
  `(("pyright.disableLanguageServices" lsp-pyright-disable-language-services t)
    ("pyright.disableOrganizeImports" lsp-pyright-disable-organize-imports t)
+   ("python.analysis.autoImportCompletions" lsp-pyright-auto-import-completions t)
+   ("python.analysis.typeshedPaths" lsp-pyright-typeshed-paths)
+   ("python.analysis.stubPath" lsp-pyright-stub-path)
    ("python.analysis.useLibraryCodeForTypes" lsp-pyright-use-library-code-for-types t)
    ("python.analysis.diagnosticMode" lsp-pyright-diagnostic-mode)
    ("python.analysis.typeCheckingMode" lsp-pyright-typechecking-mode)
    ("python.analysis.logLevel" lsp-pyright-log-level)
    ("python.analysis.autoSearchPaths" lsp-pyright-auto-search-paths)
    ("python.analysis.extraPaths" lsp-pyright-extra-paths)
-   ("python.pythonPath" lsp-pyright-locate-python)))
+   ("python.pythonPath" lsp-pyright-locate-python)
+   ("python.venvPath" lsp-pyright-venv-path)))
 
 (lsp-dependency 'pyright
                 '(:system "pyright-langserver")
                 '(:npm :package "pyright"
                        :path "pyright-langserver"))
 
+;; It additionally handles cases such as section is nested path. i.e: "python.analysis"
+(lsp-defun lsp-pyright--workspace-configuration (_workspace (&ConfigurationParams :items))
+  "Get section configuration.
+PARAMS are the `workspace/configuration' request params"
+  (->> items
+       (-map (-lambda ((&ConfigurationItem :section?))
+               (apply 'ht-get* (append (list (lsp-configuration-section section?))
+                                       (split-string section? "\\.")))))
+       (apply #'vector)))
+
 (lsp-register-client
  (make-lsp-client
   :new-connection (lsp-stdio-connection (lambda ()
                                           (cons (lsp-package-path 'pyright)
                                                 lsp-pyright-langserver-command-args)))
   :major-modes '(python-mode)
   :server-id 'pyright
-  :multi-root lsp-pyright-multi-root
+  :multi-root (lambda () lsp-pyright-multi-root)
   :priority 3
-  :initialization-options (lambda () (ht-merge (lsp-configuration-section "pyright")
-                                          (lsp-configuration-section "python")))
   :initialized-fn (lambda (workspace)
                     (with-lsp-workspace workspace
+                      ;; we send empty settings initially, LSP server will ask for the
+                      ;; configuration of each workspace folder later separately
                       (lsp--set-configuration
-                       (ht-merge (lsp-configuration-section "pyright")
-                                 (lsp-configuration-section "python")))))
+                       (make-hash-table :test 'equal))))
```

Comment:
We need to send this empty configuration, otherwise `pyright-langserver` will never request for `workspace/configuration`. (This behavior is same as VSCode Pyright extension).